pinot-core/src/main/java/org/apache/pinot/core/segment/processing/partitioner/PartitionerFactory.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.segment.processing.partitioner;
+
+import com.google.common.base.Preconditions;
+
+
+/**
+ * Factory for Partitioner and PartitionFilter
+ */
+public final class PartitionerFactory {
+
+  private PartitionerFactory() {
+
+  }
+
+  public enum PartitionerType {
+    NO_OP, ROW_HASH, COLUMN_VALUE, TRANSFORM_FUNCTION, TABLE_PARTITION_CONFIG
+  }
+
+  /**
+   * Construct a Partitioner using the PartitioningConfig
+   */
+  public static Partitioner getPartitioner(PartitioningConfig config) {
+
+    Partitioner partitioner = null;
+    switch (config.getPartitionerType()) {
+      case NO_OP:
+        partitioner = new NoOpPartitioner();
+        break;
+      case ROW_HASH:
```

Comment:
I feel ROW_HASH is not really useful. Maybe ROUND_ROBIN is enough?

Reply:
The intention of ROW_HASH is to have a partitioner which lets you set number of partitions. The existing SegmentConverter supports only this type of partitioning. Agree that ROW_HASH is not very indicative of the intent.
How about NUM_PARTITIONS or FIXED_NUM_PARTITIONS

Reply:
I understand the intention for this partitioner. My suggestion is to not use the hashcode of the record, but simply do round robin on the configured `_numPartitions` which guarantees the records for each partition are balanced. This is also much cheaper comparing to the hashcode based partitioner.